outlines/generate/generator.py

```diff
@@ -284,5 +284,5 @@ def bias_logits(logits: torch.Tensor, allowed_token_ids: List) -> torch.Tensor:
     """
     biased_logits = torch.full(logits.shape, -math.inf, device=logits.device)
```

Comment:
Thank you for opening a PR! Could we create `biased_logits` with `dtype=logits.dtype` instead? Or better, use [torch.full_like](https://pytorch.org/docs/stable/generated/torch.full_like.html)

Reply:
Ahh ok, if bias_logits can stay in half format, than this is a nicer solution. I wasn't sure what happens upstream.